cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -585,7 +585,7 @@ func (csr *ClusterStateRegistry) updateReadinessStats(currentTime time.Time) {
 			klog.Warningf("Failed to get nodegroup for %s: %v", unregistered.Node.Name, errNg)
 			continue
 		}
-		if nodeGroup == nil {
+		if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
```

Comment:
Did you actually encounter this? I think the cloudprovider which returned nil wrapped in an interface should be fixed. Not this place.

Reply:
I  actually encounter this. And i found other place judied the value is nil except here.